### PR TITLE
CEDS-4633 Amend migration that is throwing exception so that all subm…

### PR DESCRIPTION
**Pull request by magicaljellybeans**
…issions are migrated

### CODE REVIEW

**Review by udinobi — APPROVED** [review]

File: app/uk/gov/hmrc/exports/controllers/CustomHealthController.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class CustomHealthController @Inject() (
+  migrationStatus: MigrationStatus,
+  configuration: Configuration,
+  environment: Environment,
+  controllerComponents: ControllerComponents
+) extends HealthController(configuration, environment, controllerComponents) {
+
+  override def ping: Action[AnyContent] =
+    if (migrationStatus.isSuccess) {
+      super.ping
```
Comment (glidester) [inline app/uk/gov/hmrc/exports/controllers/CustomHealthController.scala:35]: Looking at the standard `HealthRoutes` class it has this comment "Audit/Auth/Logging filters rely on HandlerDef configuration (controller) in order to attach it to the request, we must extend GeneratedRouter and use the invoker tricks.."
Just worried that our directly calling `healthController.ping` won't be picked up by the filters :thinking:

File: conf/prod.routes [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Declare our custom health check first to ensure the deployment fails if migrations fail
+GET     /ping/ping        uk.gov.hmrc.exports.controllers.CustomHealthController.ping()
```
Comment (glidester) [inline conf/prod.routes:2]: Does defining this before the `health.Routes` override the standard one (I would have assumed you had to define it after to override, but I may be wrong) :thinking:

**Review by glidester — COMMENTED** [review]
Just two concerns about the implementation. The approach looks good, just need to ensure we don't break any standard functionality along the way :grinning:

File: conf/prod.routes (hunk 2 above)
Comment (magicaljellybeans) [reply conf/prod.routes:2]: First route declared has priority https://www.playframework.com/documentation/2.8.x/ScalaRouting#Routing-priority

File: app/uk/gov/hmrc/exports/controllers/CustomHealthController.scala (hunk 1 above)
Comment (magicaljellybeans) [reply app/uk/gov/hmrc/exports/controllers/CustomHealthController.scala:35]: Hmm, perhaps I need a CustomHealthRoutes that invokes our version of ping too

**Review by udinobi — APPROVED** [review]

**Pull request merged**